mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+    is returned.
+
+    Parameters
+    ----------
+    sfreq : float
+        Sampling frequency.
+    duration : float
+        Time in seconds.
+    n_epochs : int
+        Number of epochs to be generated.
+    f_phase : float
+        Phase modulating frequency in Hz (eg. 8).
+    f_amplitude : float
+        Amplitude modulated frequency eg. gamma.
+    sigma : float
+        Standard deviation of gaussian window used. The gaussian window
+        determines various levels of modulation in the trials.
+        Default value is 5.0.
+    max_amp : float
+        Constant that determines maximum amplitude of both signals.
+        Default value is 1.0.
+    random_state : None | int | instance of np.random.RandomState
+        np.random.RandomState to initialize the noise generation.
+        Default value is None.
+    mean : float
+        Mean of the Gaussian distribution (np.random.normal)
+        used to generate white noise. Default value is 0.0.
+    std : float
+        Standard deviation of the Gaussian distribution
+        (np.random.normal) used to generate white noise.
+        Default value is 0.2.
+
+    Returns
+    -------
+    pac_signal : ndarray, shape (n_epochs, n_times)
+        Phase amplitude coupled signal.
+    """
+
+    times = np.linspace(0, duration, sfreq * duration)
+    n_times = len(times)
+
+    if n_epochs <= 1:
+        raise RuntimeError('No. of trials too low, please provide atleast 2.')
+
+    rng = check_random_state(random_state)
+    # generate white noise
+    white_noise = rng.normal(mean, std, n_times * n_epochs)
+    white_noise = np.reshape(white_noise, (n_epochs, n_times))
+
+    if sigma == 0:
+        raise ValueError('Sigma cannot be 0. Please choose a higher value.')
+
+    # make a gaussian window
+    win = gaussian(n_epochs, sigma, sym=False)
+
+    if np.max(win) == np.min(win):
+        raise RuntimeError('win values lead to divide by 0.')
+
+    # normalize the gaussian window
+    win = (win - np.min(win)) / (np.max(win) - np.min(win))
+
+    # Construct the signal with many levels of modulation
+    pac_signal = np.zeros((n_epochs, n_times))
+
+    for sig in range(n_epochs):
+        # Generate amplitude envelope of modulated signal
+        amp_fa = (max_amp * ((1. - win[sig]) *
+                             np.sin(2 * np.pi * f_phase * times)
+                             + 1. + win[sig]) / 2.)
+        # Generate the phase amplitude coupled signal
+        pac_signal[sig] = (amp_fa * np.sin(2. * np.pi * f_amplitude * times) +
+                           max_amp * np.sin(2. * np.pi * f_phase * times) +
+                           white_noise[sig])
+
+    if len(pac_signal) != n_epochs:
+        raise RuntimeError('Data not equal to epochs.')
+
+    # mix the trials
+    order = np.argsort(rng.randn(len(pac_signal)))
+    pac_signal = pac_signal[order, :]
+
+    return pac_signal
+
+
+def cross_frequency_coupling(data, sfreq, phase_freq, n_cycles,
```

Comment:
I think this should be renamed to something else. It doesn't really compute any cross frequency coupling metrics, which I think is what the name of the function suggests. This function returns the evoked raw signal and the evoked spectral signal that is time-locked to the phase peak. I think that a better function name would be `phase_locked_amplitude` or something like this. Thoughts?